src/d3a/models/market/blockchain_interface.py

```diff
@@ -59,11 +65,88 @@ def bc_listener(self):
         pass
 
 
+class SubstrateBlockchainInterface():
+    def __init__(self):
+        self.contracts = {'main': main_address}
+        self.substrate = SubstrateInterface(
+            url="ws://127.0.0.1:9944",
+            address_type=42,
+            type_registry_preset='default'
+        )
+
+    def simple_contract_call(self, contract_address, function, path):
+        data = get_function_metadata(function, path)
+        call = self.substrate.compose_call(
+            call_module='Contracts',
+            call_function='call',
+            call_params={
+                'dest': contract_address,
+                'value': 1 * 10**18,
+                'gas_limit': 1000000000000,
+                'data': data
+            }
+        )
+        return call
+
+    def storage_contract_call(self, contract_address, function, path):
+        trade_id = random.randint(10, 16)
+        print('TRADE ID {}'.format(trade_id))
+
+        data = get_function_metadata(function, path) + str(trade_id) + \
+            address_to_hex(BOB_ADDRESS) + address_to_hex(ALICE_ADDRESS) + \
+            swap_byte_order(hex2(test_value)) + '000000000000000000' + hex2(test_rate)
+        print('DATA {}'.format(data))
+        call = self.substrate.compose_call(
+            call_module='Contracts',
+            call_function='call',
+            call_params={
+                'dest': contract_address,
+                'value': 1 * 10**18,
+                'gas_limit': 1000000000000,
+                'data': data
+            }
+        )
```

Comment:
Please abstract it into a method, I see a code repetition [here](https://github.com/gridsingularity/d3a/pull/883/files#diff-797fb81c9cd9bd008da4c6892e371a50R79-R88)

Reply:
Good point, will do

Reply:
Done